testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -321,14 +333,38 @@ impl BlockMinerThread {
         for (_slot, signer_message) in signer_messages {
             match signer_message {
                 SignerMessage::Transactions(transactions) => {
-                    // TODO: filter out transactons that are not valid and that do not come from the signers
-                    // TODO: move this filter function from stacks-signer and make it globally available perhaps?
-                    transactions_to_include.extend(transactions);
+                    for transaction in transactions {
+                        let address = transaction.origin_address();
+                        let nonce = transaction.get_origin_nonce();
+                        if !addresses.contains(&address) {
+                            test_debug!("Miner: ignoring transaction ({:?}) with nonce {nonce} from address {address}", transaction.txid());
+                            continue;
+                        }
+
+                        let cur_nonce = chainstate
+                            .with_read_only_clarity_tx(
+                                &sortdb.index_conn(),
+                                &self.parent_tenure_id,
+                                |clarity_tx| {
+                                    clarity_tx.with_clarity_db_readonly(|clarity_db| {
+                                        clarity_db.get_account_nonce(&address.into()).unwrap_or(0)
+                                    })
+                                },
+                            )
+                            .unwrap_or(0);
+
+                        if cur_nonce > nonce {
```

Comment:
No, it could be that there are more transactions than just one. One might be in the mempool from the same account (Suppose the current nonce is 1. Transaction 1 is in the mempool, transaction 2 is in stackerdb...both are valid. We only know transactions with a nonce LESS than the current are invalid as they will never be approved.)